docs/source/developer/translation.rst

```diff
@@ -0,0 +1,128 @@
+Translation
```

Comment:
Maybe add common translation examples, like how to translate words that may or not be plural, or just add a link to the docs for it